File: apps/web/src/lib/dbPaths.ts
const DEFAULT_TENANT_ID = process.env.NEXT_PUBLIC_TENANT_ID || null;

export const dbPaths = {
  sessions: (tenantId: string | null = DEFAULT_TENANT_ID) => 
    tenantId ? `tenants/${tenantId}/sessions` : `csm_sessions`,
    
  sessionDoc: (id: string, tenantId: string | null = DEFAULT_TENANT_ID) => 
    tenantId ? `tenants/${tenantId}/sessions/${id}` : `csm_sessions/${id}`,
    
  articles: (tenantId: string | null = DEFAULT_TENANT_ID) => 
    tenantId ? `tenants/${tenantId}/articles` : `articles`,
    
  articleDoc: (id: string, tenantId: string | null = DEFAULT_TENANT_ID) => 
    tenantId ? `tenants/${tenantId}/articles/${id}` : `articles/${id}`,
    
  configs: (tenantId: string | null = DEFAULT_TENANT_ID) => 
    tenantId ? `tenants/${tenantId}/agent_configurations` : `agent_configurations`,
    
  configDoc: (agentName: string, tenantId: string | null = DEFAULT_TENANT_ID) => 
    tenantId ? `tenants/${tenantId}/agent_configurations/${agentName}` : `agent_configurations/${agentName}`,
    
  apiKeysDoc: (tenantId: string | null = DEFAULT_TENANT_ID) => 
    tenantId ? `tenants/${tenantId}/api_keys/keys` : `agent_configurations/api_keys`,
    
  socialQueue: (tenantId: string | null = DEFAULT_TENANT_ID) => 
    tenantId ? `tenants/${tenantId}/social_queue` : `social_queue`,
    
  socialQueueDoc: (id: string, tenantId: string | null = DEFAULT_TENANT_ID) =>
    tenantId ? `tenants/${tenantId}/social_queue/${id}` : `social_queue/${id}`,

  usageLogs: (tenantId: string | null = DEFAULT_TENANT_ID) =>
    tenantId ? `tenants/${tenantId}/usage_logs` : `usage_logs`,

  videoTemplates: (tenantId: string | null = DEFAULT_TENANT_ID) =>
    tenantId ? `tenants/${tenantId}/video_templates` : `video_templates`,

  videoTemplateDoc: (id: string, tenantId: string | null = DEFAULT_TENANT_ID) =>
    tenantId ? `tenants/${tenantId}/video_templates/${id}` : `video_templates/${id}`,

  avatarsDoc: (tenantId: string | null = DEFAULT_TENANT_ID) =>
    tenantId ? `tenants/${tenantId}/agent_configurations/avatars` : `agent_configurations/avatars`,
};

File: agents/cmo_agent/db_paths.py
from contextvars import ContextVar
from typing import Optional

tenant_id_var: ContextVar[Optional[str]] = ContextVar("tenant_id", default=None)
session_id_var: ContextVar[Optional[str]] = ContextVar("session_id", default=None)

def get_tenant_id() -> Optional[str]:
    return tenant_id_var.get()

def set_tenant_id(tenant_id: Optional[str]):
    tenant_id_var.set(tenant_id)

def get_session_id() -> Optional[str]:
    return session_id_var.get()

def set_session_id(session_id: Optional[str]):
    session_id_var.set(session_id)

def get_sessions_path() -> str:
    tid = get_tenant_id()
    return f"tenants/{tid}/sessions" if tid else "csm_sessions"

def get_session_doc_path(session_id: str) -> str:
    tid = get_tenant_id()
    return f"tenants/{tid}/sessions/{session_id}" if tid else f"csm_sessions/{session_id}"

def get_articles_path() -> str:
    tid = get_tenant_id()
    return f"tenants/{tid}/articles" if tid else "articles"

def get_article_doc_path(article_id: str) -> str:
    tid = get_tenant_id()
    return f"tenants/{tid}/articles/{article_id}" if tid else f"articles/{article_id}"

def get_config_doc_path(agent_name: str) -> str:
    tid = get_tenant_id()
    return f"tenants/{tid}/agent_configurations/{agent_name}" if tid else f"agent_configurations/{agent_name}"

def get_configs_path() -> str:
    tid = get_tenant_id()
    return f"tenants/{tid}/agent_configurations" if tid else "agent_configurations"

def get_api_keys_doc_path() -> str:
    tid = get_tenant_id()
    return f"tenants/{tid}/api_keys/keys" if tid else "agent_configurations/api_keys"

def get_social_queue_path() -> str:
    tid = get_tenant_id()
    return f"tenants/{tid}/social_queue" if tid else "social_queue"

def get_usage_logs_path() -> str:
    tid = get_tenant_id()
    return f"tenants/{tid}/usage_logs" if tid else "usage_logs"

def get_local_project_dir(session_id: str) -> str:
    import os
    base_dir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
    projects_dir = os.path.join(base_dir, "projects")
    project_dir = os.path.join(projects_dir, session_id)
    return project_dir
